### PR TITLE
Created Readme file for the project codebase

### PR DESCRIPTION
**Pull request by KavsHub**
Readme for the project has been made. (Issue #52)
This file gives an overview of our git structure, project setup and current release user stories.
Please let me know if there's anything else I need to add before I merge it into the develop branch

### CODE REVIEW

**Comment by KavsHub** [comment]
**Since Comments use the same markup, I just put the text into here so you guys can see it easier.**

# CASS-Degrees-Code
Primary Codebase for the TechLauncher CASS Degrees project

For our landing page please go to: https://github.com/cass-degrees/CASS-Degrees

## Our Git Branching Structure
In this project we will have two main branches: **Master** and **Develop**

The **Master** branch will only contain completed and working releases that have addressed all the user stories of the relevant release. 
Merges to this branch will only come from the develop branch once a release in develop has been completed and finalised.

The **Develop** branch is the main working branch which will have completed and working user stories for the relevant release that the team is working on.
No one is to directly commit to this branch, it is updated by merges from branches created for the relevant user story or feature that a developer is working on.

The idea is that once someone completes a feature or user story, they will create a new branch with their code (with an appropriate name) and request a *Pull Request* into
the develop branch.

**Pull Requests** are reviewed by the projects team members, the rule is that 2 people must approve of the feature or user story before it can
be merged into the develop branch. Alternatively, if it has been seen by the project team members and has been open for more than 2 days, then the
developer can merge into the develop branch, given that they have tested and ensured that their code is functional. 


Please see this document for a full overview and detailed explanation of our branching structure: https://docs.google.com/document/d/1whU7j2F_0RFJ1n1rJsn3H6UcKvLCPCmKZ2J65a7D-9g/edit

## Project Environment Setup 
Please view this link: https://docs.google.com/document/d/1AP6ycdfm0Rk04vRhcsrh3G53PR-BPytMMqTyLkMv7vQ/edit?usp=sharing

Follow the steps in the guide and you should be able to setup the environment on your own local machine.

If there are any issues with the setup process, please create an issue and we will address it as soon as possible.

## User Stories and Acceptance Criteria (MVP)
The user stories and acceptance criteria for the MVP can be found here: https://docs.google.com/document/d/1nj0gj3BS6XL3B7XeQXr2or4Vxlh1Ah5AFw33DN0ZuBs/edit?usp=sharing

These stories have been converted into issues which have been assigned to members of the project team

File: README.md
Comment (j-selby) [inline README.md]: This should be moved into this repo entirely IMO - build instructions might differ between versions, and it should be versioned directly.

File: README.md
Comment (j-selby) [inline README.md]: This isn't exactly correct - make sure to review the decision log for this

**Review by j-selby — CHANGES_REQUESTED** [review]

File: README.md
Comment (KavsHub) [reply README.md]: Fair enough but I thought it should be here for now as it is our current project environment setup guide and it will help people who want to play with our program.
In the next release, we can update this link/information if anything has changed.

File: README.md
Comment (KavsHub) [reply README.md]: Yeah I wasn't really sure about the changes, I'll update this bit

**Comment by KavsHub** [comment]
**UPDATED** Again this is for easier viewing

# CASS-Degrees-Code
Primary Codebase for the TechLauncher CASS Degrees project

For our landing page please go to: https://github.com/cass-degrees/CASS-Degrees

## Our Git Branching Structure
In this project we will have two main branches: **Master** and **Develop**

The **Master** branch will only contain completed and working releases that have addressed all the user stories of the relevant release. 
Merges to this branch will only come from the develop branch once a release in develop has been completed and finalised.

The **Develop** branch is the main working branch which will have completed and working user stories for the relevant release that the team is working on.
No one is to directly commit to this branch, it is updated by merges from branches created for the relevant user story or feature that a developer is working on.

The idea is that once someone completes a feature or user story, they will create a new branch with their code (with an appropriate name) and request a *Pull Request* into
the develop branch.

**Pull Requests** are reviewed by the projects team members, the rule is that 2 people must approve of the feature or user story before it can be merged into the develop branch. 
Alternatively, if a day or more has past then only 1 approval is required for the pull request. This is to ensure the lively hood of our pull requests so that they don't go stale and reduce productivity of other people’s sections who may depend on a pull request.

Please see this document for a full overview and detailed explanation of our branching structure: https://docs.google.com/document/d/1whU7j2F_0RFJ1n1rJsn3H6UcKvLCPCmKZ2J65a7D-9g/edit

## Project Environment Setup 
Please view this link: https://docs.google.com/document/d/1AP6ycdfm0Rk04vRhcsrh3G53PR-BPytMMqTyLkMv7vQ/edit?usp=sharing

Follow the steps in the guide and you should be able to setup the environment on your own local machine.

If there are any issues with the setup process, please create an issue and we will address it as soon as possible.

## User Stories and Acceptance Criteria (MVP)
The user stories and acceptance criteria for the MVP can be found here: https://docs.google.com/document/d/1nj0gj3BS6XL3B7XeQXr2or4Vxlh1Ah5AFw33DN0ZuBs/edit?usp=sharing

These stories have been converted into issues which have been assigned to members of the project team

File: README.md
Comment (KavsHub) [reply README.md]: I just realised I misinterpreted what you said. 
It would be good to have the instructions in the readme but I feel like the current instructions are too long and having it isolated is good for if we change things as we don't need to update the readme and the setup document. We'll see what everyone else thinks.

Comment (j-selby) [reply README.md]: If the instructions are too long for the readme, chuck them in a separate markdown file. They are intrinsically attached to the codebase at particular points.

File: README.md
Comment (j-selby) [inline README.md]: Title doesn't need to be dashed :P

Comment (KavsHub) [reply README.md]: Haha the title was actually auto generated by GitHub. Didn't even notice it :')

**Comment by KavsHub** [comment]
the new file SETUP.md contains the project setup guide now.
Here is the file in a comment for convenience
**Note: the list notation doesn't seem to work here pls ignore**
**Another note: I found some spelling mistakes... I'll commit them once theres more changes**

# CASS Degrees Code Setup Guide
This document will guide you through the steps to setup your own local environment for the current CASS Degrees project release

For the main readme please go to go to README.md file located in this repository. 

## How to run our Django app (Windows)
*Created by Daniel Jang*

**NOTE: This guide is focused for Windows users, but the basic steps for other OS is similar (Install Python -> install PostgreSQL -> Create DB -> set up PyCharm -> run)**



1. Install Python version 3.7.2 from this link (Mac users can also choose to do apt-get instead):
https://www.python.org/downloads/release/python-372/
Select the option to add Python to your PATH on Windows

2. Install PostgreSQL server (more for Windows users, Mac users can refer to this for steps 2-4): https://medium.com/@viviennediegoencarnacion/getting-started-with-postgresql-on-mac-e6a5f48ee399):
https://www.enterprisedb.com/downloads/postgres-postgresql-downloads

3. During PostgreSQL installation, leave everything default until you get to the password step:
    3. Password: postgres
    3. Leave everything after that as default, but there is no need to install StackBuilder.

4. Open pgAdmin4.exe found in “C:\Program Files\PostgreSQL\11\pgAdmin 4\bin” (or wherever the program was installed).
    4. If PostgreSQL is not present in Program Files, try Program Files (x86).

5. Create new database for our Django app: 
    5. On the left side, you should see a dropdown menu called Servers (1).
    5. Open that by double clicking, and you will see PostgreSQL. Open that and enter the password we set in step 3a.
    5. Now you will see “Databases (1)” underneath PostgreSQL. Right click on it and click “Create > Database…”
    5. Call the new DB “cassdegrees” (without quotes). Make sure the owner is set as “postgres” and then click save.

6. Install PyCharm Professional and activate it using your educational JetBrains account (or start a free trial):
    6. https://www.jetbrains.com/pycharm/download/#section=windows

7. Open our project by checking out from version control and selecting git.
    7. Login to GitHub and select our project (CASS-Degrees-Code) and choose directory where you want to save it if you want to change from default.
    7. Ensure that you are in the right git branch by checking the branch in the bottom right of PyCharm. It should initially say “Git: master”. Click on it and change it to a branch you want to actually see.
        7. If the branch isn’t there, go to “VCS > Git > Fetch” and try again.
        7. If the branch is there, click on it and then click “Checkout As...” and then leave the default as is and click OK.
        7. Don’t add \.idea\vcs.xml to git ***** (UNKNOWN DOWNSIDES, will confirm later if step is necessary. Don’t check “Remember” for now)

8. Ensure that PyCharm has the correct project interpreter:
    8. Go to “File > Settings > Project: CASS-Degrees-Code > Project Interpreter” and check if interpreter is the Python version you installed and is not invalid.
    8. If invalid, click on the Python 3.6 (or whatever it says in red) and click on “Show All…”
    8. Remove all invalid ones.
    8. On the top right of the “Project Interpreters” window, click on the + button.
    8. For the “Base interpreter” field, find the correct python.exe file and click OK:
        8. Mine was: “C:\Users\<USERNAME>\AppData\Local\Programs\Python\Python37\python.exe” but yours may be in Program Files.
        8. After virtual environment is made and the correct Python interpreter appears in “Project Interpreters”, click OK and then OK again to close Settings.

9. Click “Add Configuration” on top right of PyCharm to add Django support:
    9. On the left, click the “+” and click on “Django Server”
    9. If you see an error saying “Error: Please enable Django support for the project” on the bottom of the window, click on the “Fix” button.
        9. Tick “Enable Django Support”.
        9. For Django project root, find the “cassdegrees” folder within the cloned git repository and click OK (NOT THE cassdegrees WITHIN cassdegrees FOLDER!!).
        9. For Settings, find the “settings.py” under “cassdegrees” folder found within the parent “cassdegrees” folder.
        9. Rest can be left blank.
        9. Click OK.
    9. Name your configuration to whatever you want, I just called it “cassdegrees”.
    9. If you see an error saying “Error: Django is not importable in this environment”
        9. Open  “cassdegrees\requirements.txt” in pycharm
        9. If you don’t have the requirements installed, pycharm will prompt you with a yellow bar at the top of the file.
        9. Click “install packages”
    9. Tick “Run browser” if you want a new window to automatically pop up.

10. At some point, a prompt will appear that requests connection to the database. Fill in the required fields (e.g. passwords), test connection and if successful, press OK.

11. If your models for the database has been altered (code in models.py), make sure you perform the migrations. This is necessary for the tables in our PostgreSQL database to be updated.
Go to terminal in PyCharm at bottom left ish, change directory so that manage.py is in your working directory, then run:
python manage.py makemigrations
python manage.py migrate

12. Now, try running the Django app by clicking the green triangle on the top right.

13. Now, you should be greeted with Django’s main page (with a flying rocket) which says that your installation is successful. From this point on, all you need to do is to switch branches as necessary and then click the green arrow.

14. To use the administrator page (127.0.0.1:8000/admin/), add yourself as the superuser:
    14. python manage.py createsuperuser
    14. You can set any value for username, email or password. Remember it though so you can log into the admin website. I did u: admin and pw: admin.

**If you made it this far then *CONGRATULATIONS!* you know have a working local environment to play with!**

**If you did and it doesn't work... then please create an issue or raise it to one of our project members and we will
try to assist you as soon as possible to the best of our abilities!**

**Review by aidengeorge97 — APPROVED** [review]
With the addition of SETUP.md I believe this request is appropriate to be merged. 

Edit: I just noticed you misspelled "livelihood" as "lively hood" and "passed" as "past" in the readme, but if you change these it should be fine.

**Pull request merged**
